### ISSUE TITLE
Events?

### ISSUE THREAD

**Issue by broberson** [feathersjs/feathers-mongoose#27]
The [Feathers docs](http://feathersjs.com/docs/#toc18) state:

> Any registered service will be automatically turned into an event emitter that emits events when a resource has changed, that is a `create`, `update` or `remove` service call returned successfully.

I'm not seeing those events from feathers-mongoose services, nor the ability to listen for them; they don't seem to be EventEmitters at all. Is that intentional, or am I just missing something obvious?

**Comment by broberson** [comment]
Alright, so if (after mounting the MongooseService as middleware with `app.use('/route', mongooseService)`) I grab a reference to that middleware using `app.service('/route')`, the thing I get back does seem to implement an EventEmitter interface.

Still, it doesn't actually seem to emit `create`, `update`, or `remove` events. Should it?

**Comment by daffl** [comment]
As long as you bind to the events before anybody calls a service method you should get the events mentioned. For your example this should work:

```js
app.service('route').on('created', function(route) {
  console.log('Route created', route);
});

app.service('route').create({
  url: '/something'
}, function() {});

// The same for POST to /route or calling
// socket.emit('route::create', { url: '/something' });
```

Note: The empty `function(){}` callback is just necessary because of a bug at https://github.com/feathersjs/feathers/issues/150 which will be fixed with the 1.2 release.

**Comment by broberson** [comment]
Thanks David, that's probably where I've gone astray.
